gui/system/migrations/0030_ip_fields.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import unicode_literals
+
+
+import json
+
+
+from django.db import migrations
+from freenasUI.freeadmin.models.fields import ListField
+
+
+def ip_update(apps, schema_editor):
+
+    settings_obj = apps.get_model('system', 'settings').objects.order_by('-id')[0]
+    for key in ['stg_guiaddress', 'stg_guiv6address']:
+        if getattr(settings_obj, key):
+            setattr(settings_obj, key, json.dumps([getattr(settings_obj, key)]))
```

Comment:
Why do you need to use json.dumps here? Isnt ListField supposed to handle it?

Reply:
This is done before the migration to change the value of the field from `0.0.0.0` to `["0.0.0.0"]` in the way ListField expects. If the value found from db is a list, listfield uses the `json.loads` method to load the value and this will fail the if the value isn't updated

Reply:
```
    def to_python(self, value):
        if not value:
            return []
        if isinstance(value, str):
            return json.loads(value)
        return value
```